### PR TITLE
DOC Document breaking changes for updating resources dir

### PR DESCRIPTION
**Pull request by GuySartorelli**
**NOTE** This PR assumes that https://github.com/silverstripe/developer-docs/pull/47 will be merged, and so it updates the `03_Upgrading/02_Upgrading_your_project.md` file from that PR.

The actual format of the upgrading docs can be sorted out closer to release - for now this serves as documentation of what breaking changes have been made.

## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10399

### CODE REVIEW

**Review by emteknetnz — APPROVED** [review]

File: en/03_Upgrading/02_Upgrading_your_project.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+## Breaking changes (by module, alphabetically)
+
+### silverstripe/framework
+
+- Removed deprecated `ManifestFileFinder::RESOURCES_DIR`
+- Changed the default for the `RESOURCES_DIR` const to "_resources"
+  - This can still be customised using `extra.resources-dir` in your composer.json file ([see relevant docs](developer_guides/templates/requirements/#configuring-your-project-exposed-folders))
+  - If your composer.json file has its `extra.resources-dir` key set to `_resources`, you can remove that now.
+  - If your composer.json file already does not have a `extra.resources-dir` key and you want to keep your resources in the `resources` directory, you can set `extra.resources-dir` to "resources".
```
Comment (emteknetnz) [inline en/03_Upgrading/02_Upgrading_your_project.md]:
```suggestion
  - If your composer.json file already does not have an `extra.resources-dir` key and you want to keep your resources in the `resources` directory, you can set `extra.resources-dir` to "resources".
```

File: en/03_Upgrading/02_Upgrading_your_project.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  - If your composer.json file already does not have a `extra.resources-dir` key and you want to use the new default `_resources` directory, you may need to check that your code and templates don't assume the directory name for those resources. In your templates it is preferred to [use `$resourePath()` or `$resourceURL()`](developer_guides/templates/requirements/#direct-resource-urls) to get paths for resources.
```
Comment (emteknetnz) [inline en/03_Upgrading/02_Upgrading_your_project.md]:
```suggestion
  - If your composer.json file already does not have an `extra.resources-dir` key and you want to use the new default `_resources` directory, you may need to check that your code and templates don't assume the directory name for those resources. In your templates it is preferred to [use `$resourePath()` or `$resourceURL()`](developer_guides/templates/requirements/#direct-resource-urls) to get paths for resources.
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Very minor grammar changes

File: en/03_Upgrading/02_Upgrading_your_project.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/03_Upgrading/02_Upgrading_your_project.md]: done

File: en/03_Upgrading/02_Upgrading_your_project.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/03_Upgrading/02_Upgrading_your_project.md]: done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**